server/src/test/java/io/crate/planner/UnionPlannerTest.java

```diff
@@ -265,4 +269,41 @@ public void test_pruneOutputsExcept_can_handle_duplicates() {
         );
     }
 
+    @Test
+    public void test_union_works_when_both_sides_have_no_outputs_to_keep_on_prunning() {
+        LogicalPlan plan = e.logicalPlan(
+            "SELECT count(*) FROM (SELECT id FROM users UNION ALL SELECT 1 as renamed) t");
+
+        assertThat(plan).isEqualTo("""
+            HashAggregate[count(*)]
+              └ Rename[] AS t
+                └ Union[]
+                  ├ Collect[doc.users | [] | true]
+                  └ TableFunction[empty_row | [] | true]
+            """
+        );
+
+        ExecutionPlan executionPlan = plan.build(
```

Comment:
The above logical plan assertions already ensures that there is no Eval on top of the table function with an output. Why do we need the assertions on the execution plan?

Reply:
I removed part with the execution plan